File: GhostLeg.java
import java.util.Scanner;

public class GhostLeg {
    
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        int n = sc.nextInt();
        int m = sc.nextInt();

        int[] rungs = new int[m];
        for (int i = 0; i < m; i++) rungs[i] = sc.nextInt();

        int[] perm = new int[n + 1];

        for (int i = 1; i <= n; i++) {
            int j = i;

            for (int r : rungs) {
                if (r == j) {
                    j++;
                } else if (r == j - 1) {
                    j--;
                }
            }

            perm[j] = i;
        }

        for (int i = 1; i <= n; i++) {
            System.out.println(perm[i]);
        }
    }
}

File: ghost_leg.py
n, m = [int(i) for i in input().split()]
rungs = [int(input()) for i in range(m)]
perm = [0] * (n + 1)

for i in range(1, n + 1):
    j = i

    for r in rungs:
        if r == j:
            j += 1
        elif r == j - 1:
            j -= 1
    
    perm[j] = i

for p in perm[1:]: print(p)